### PR TITLE
Run AKS template test without playwright validation

### LINKED ISSUE #3394: [Issue] azd cli github action for aks cluster: kubelogin not found in path
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
Run `azd version` and copy and paste the output here:
image: mcr.microsoft.com/azure-dev-cli-apps:latest

**Describe the bug**
Running azd in a github action with the latest version.  After provisioning the aks cluster, it fails with the error message:
  ```
(✓) Done: AKS Managed Cluster: aks-azdaks01
ERROR: deployment failed: failed invoking event handlers for 'postprovision', converting kubeconfig: exec: "kubelogin": executable file not found in $PATH,converting kubeconfig: exec: "kubelogin": executable file not found in $PATH
Error: Process completed with exit code 1.
```

**To Reproduce**
Run AZD Up on the cqgh branch of 
https://github.com/Azure-Samples/ratings-nodejs-azd-akslza

**Expected behavior**
Expect for kubelogin to be available in the path on the azd image.

### PR DESCRIPTION
**Pull request by wbreza**
Resolves #3394
Adds the following tools to the `azd` container image

- [x] Helm
- [x] Kustomize
- [x] Kubelogin  

Updates AKS template tests to run without playwright validation tests.
In our test tenant the AKS cluster is not internet reachable from DevOps because of subscription wide networking policies and causes the template tests to fail.

This change would allow the azd init, up flow to be validated at least and would only bypass the playwright tests.

### CODE REVIEW

File: templates/cspell-templates.txt [hunk 1]
```diff
@@ -81,4 +81,5 @@ PSQLADMINNAME
 PSQLADMINPASSWORD
 PSQLUSERNAME
 PSQLUSERPASSWORD
-DBSERVER
\ No newline at end of file
+DBSERVER
+test
```
Comment (wbreza) [inline templates/cspell-templates.txt:85]: This is here just to force a global change to the templates.

File: templates/todo/common/tests/README.md [hunk 2]
```diff
@@ -26,4 +26,6 @@ Add the `--debug` flag to run with debugging enabled. You can find out more info
 npx playwright test --debug
 ```
 
-More debugging references: https://playwright.dev/docs/debug and https://playwright.dev/docs/trace-viewer
\ No newline at end of file
+More debugging references: https://playwright.dev/docs/debug and https://playwright.dev/docs/trace-viewer
+
+force change
```
Comment (wbreza) [inline templates/todo/common/tests/README.md:31]: Only here to force a change across all the templates.

File: eng/pipelines/templates/steps/template-test-run-job.yml [hunk 3]
```diff
@@ -83,6 +83,7 @@ steps:
               BRANCH_NAME=$(TemplateBranchName)
               ENV_NAME_PREFIX=$(EnvPrefixName)
               TEMPLATE_NAME=$(TemplateName)
+              VALIDATE=$(TemplateName)
```
Comment (weikanglim) [inline eng/pipelines/templates/steps/template-test-run-job.yml:86]: Curious... Is this line still needed?

File: templates/tests/test-templates.sh [hunk 4]
```diff
@@ -98,8 +102,8 @@ function deployTemplate {
         azd env new "$3" --subscription "$4" --location "$5" --no-prompt
     fi
 
-    echo "Create provision preview for $3..."
-    azd provision -e "$3" --preview
+    # echo "Create provision preview for $3..."
+    # azd provision -e "$3" --preview
```
Comment (weikanglim) [inline templates/tests/test-templates.sh:106]: I think we still want to keep this?

File: templates/tests/test-templates.sh [hunk 5]
```diff
@@ -116,6 +120,11 @@ function deployTemplate {
 # $2 - The branch name
 # $3 - The environment name
 function testTemplate {
+    if [ $VALIDATE == false ]; then 
+        echo "Skipping playwright validation for $1..."
+        return
```
Comment (weikanglim) [inline templates/tests/test-templates.sh:125]: Do we know an idea of what the effort in how we can validate AKS in the long run? It is possible that we run some `az` update commands to poke through the firewall for this particular test setup?

These are non-prod resources, and it seems like we're just subjected to the default network policies that are more restrictive than what we intend for. Would be nice to have a work item capturing this so we don't forget.

**Review by weikanglim — APPROVED** [review]
Happy to see the work to start enabling AKS tests, with playwright disabled initially. Would be good to create and follow-up on whichever remaining items that can enable the playwright tests.

File: eng/pipelines/templates/steps/template-test-run-job.yml (hunk 3 above)
Comment (wbreza) [reply eng/pipelines/templates/steps/template-test-run-job.yml:86]: No, i can remove this. Thanks :)

File: templates/tests/test-templates.sh (hunk 4 above)
Comment (wbreza) [reply templates/tests/test-templates.sh:106]: There is a bug i logged regarding this.  
https://github.com/Azure/azure-dev/issues/3447

However in the template tests does preview provide any real value outside of validating the command works? If that is the intent then we can leave it in but we should address the bug mentioned above.

File: templates/tests/test-templates.sh (hunk 5 above)
Comment (wbreza) [reply templates/tests/test-templates.sh:125]: Logging internal issue @ https://github.com/Azure/azure-dev-pr/issues/1608

**Pull request merged**